mosec/coordinator.py

```diff
@@ -67,6 +70,54 @@ def __init__(
             timeout=PROTOCOL_TIMEOUT,
         )
 
+        # optional plugin features - plasma object store as shared memory
+        if shm_path != "":
+            shm_cli = ShmClient(shm_path)
+
+            def get_decoder() -> Callable:
+                d_ipc = self.worker._deserialize_ipc
+
+                def batch_decoder(batch):
+                    if STAGE_INGRESS in self.worker._stage:
+                        return [self.worker.deserialize(x) for x in batch]
+                    objects = shm_cli.get([d_ipc(x) for x in batch])
```

Comment:
Will it be better to move these two `shm` methods to the `receive` and `send` parts?

For example:

```py
def receive(self, wrapper=lambda x: x):
    # ...
    return flag, ids, wrapper(payloads)
```

Reply:
Essentially this is the same as the previous `IPCWrapper` right? No matter `shm` enabled or not, there will be a function call wrapped.

But the last commit's idea is to disable the wrapping when plugin is disabled?

Reply:
Bad example. Try this:

```py
def receive(self):
    pass

# ...
if wrapper:
    self.receive = wrapper(self.receive)
```